src/shrub/base.py

```diff
@@ -23,7 +23,10 @@ def to_map(self):
     def _add_if_defined(self, obj, prop):
         """Add the specified property to the given object if it exists."""
         value = getattr(self, prop)
-        if value:
+
+        # We can't just use the falsiness of value because false is a value
+        # with semantic meaning different from undefined in some keys.
+        if value is not None and value != [] and value != {}:
```

Comment:
I think so. This shows up in the tests. The current behavior because it tests falsiness is to not include a key for empty dicts and arrays and I didn't want to change that.